### DISCUSSION TITLE
Dynamic classification of events

### DISCUSSION THREAD

**Discussion by unode** [ActivityWatch/activitywatch#947] [Q&A]
Hi everyone,

I've been using ActivityWatch for some years now and every now and then I run into categorization ambiguity.

These are typically one of two cases:
1. A given event could be matched under more than one label - Regexp are too generic or they match something that isn't unique to a category, e.g. an acronym.
2. Window titles or entries are too general or impossible to capture in a regex without singling them out.

How do you typically address these?
I'm thinking of a solution where I would be able to dynamically assign each uncategorized event to a given category but this isn't particularly convenient.

The category builder is useful but it also suffers from that described in 1.

**Comment by BelKed** [comment]
> 1. A given event could be matched under more than one label - Regexp are too generic or they match something that isn't unique to a category, e.g. an acronym.

I would make the RegEx match only the window title if possible.

---

> 2. Window titles or entries are too general or impossible to capture in a regex without making singling them out.

If you don't mind "losing" some data, I wouldn't worry too much about too general entries.

---

Personally, I've been using ActivityWatch for more than a year now, and from what I can tell about my computer usage, I don't spend too much time on some applications that I haven't categorised yet, so I don't suffer too much from these problems...

Maybe, if you could give me some specific examples of your problems, I might be able to help you more :)

**Reply by unode** [reply]
For some categories I would like to capture the project I'm working on.

These sometimes include paths (in terminal/shells), partial filenames (e.g. in Gdocs or browser titles), meeting names (Zoom, Meet, Teams)...

Other times the windows have generic names such as "Zoom meeting" or "Some_meeting - Google docs" which are impossible to categorize under a project. Sure I could simply categorize them as "Videocall" or "Meetings" but I was hoping to be able to define something more specific.
The idea behind manually assigning them to a category is mostly to be able to get an aggregated report at the end of the day that contains (almost) no uncategorized events. 

An example of ambiguous is "GitHub" which appears in the title of all kinds of pages. Some on github.com, but many on random websites mentioning or referencing GitHub pages.

**Reply by 2e3s** [reply]
> An example of ambiguous is "GitHub" which appears in the title of all kinds of pages. Some on github.com, but many on random websites mentioning or referencing GitHub pages.

I don't see Github on github.com, anyway, the real challenge here is to intersect different buckets, that is to use the current category of `aw-watcher-web` to assign the same category to the current app and window title events, I guess? So there would be a part of uncategorized "firefox", a part of categorized "firefox".

**Comment by Jasha10** [comment]
+1 Categorizing activities on an ad-hoc basis would be a killer feature.
To echo what @unode said, sometimes "Google Docs" is the "Work" category and sometimes it is "Personal".
